### ISSUE TITLE
CRF usage and hyper parameter tuning  for chunking task with only two tags

### ISSUE THREAD

**Issue by Michael95-m** [flairNLP/flair#1595]
Currently, I am training the **chunking** (**word break**) task with only two **B and I** (beginning and intermediate) tags. The language I want to train is one of the sino-tibetan language and in that language there is **no proper word break**(only space is used for reading clearly) like Chinese and Japanese. So, begin and intermediate tags are used to predict for **each syllable** to become a word. What I would like to know is that
1. Does **CRF** need for this task?? What I understood is that for sequence labeling, CRF mainly helps to prevent invalid label sequences theoretically. In my task, only one invalid label sequence is that predicting starting tag (first syllable in sentence) is intermediate tag.  When I trained the model both with and without CRF, training with CRF achieved more F1 scores(nearly 1%). In that training, the embedding I use was **stacked** embedding of **character** and **byte pair** embedding. I don't know why does training with CRF much better for this task?
2. Secondly, to improve this chunking task, should I use more **layers** or more **hidden states**? Currently I used 256 hidden states and 2 RNN layers. 
3. Finally, for the case of sequence labeling with **syllables**, which kind of **embedding** might be better for this task?? In the future, I want to use flair embedding for this task. But currently, there is no pretrained language model for the language I want to train and currently I just start to collect the data for scratch training.

**Comment by alanakbik** [comment]
Hello @Michael95-m if CRF gives a better score, then its good for the task ;) CRF does a better decoding of the final word-level predictions so it may help with some of those B-I tags. I would not increase layers or states - If anything, I would decrease to 1 layer - but your settings seem good. You could try playing around with the mini-batch size (if yu have lots of data you could try increasing, if you have little data you could try decreasing). 
But the most important thing are the word embeddings. You could try other word embeddings if they exist for your language which has the biggest impact. Especially FlairEmbeddings would probably be very useful in this task. You could also try multilingual BERT if it supports your language.

**Comment by Michael95-m** [comment]
Thanks for your recommendation @alanakbik . Btw, I found that training BiLSTM CRF with multilingual  BERT takes very long time and its increase in accuracy is  slower than other embedding. But I found that finetuning BERT(got it from lastest branch repo) with linear layer gives better result.

**Comment by alanakbik** [comment]
Interesting - can you share the final evaluation numbers for both setups?

**Comment by Michael95-m** [comment]
Of course @alanakbik . Firstly, I was training this chunking task with bert_embedding('multilingual-cased') and 2 layers of rnn and crf. Even after 20 epochs,  the accuracy didn't exceed over 95% and one epoch takes nearly 2 hours. Then I changed my approach to finetuning with bert_embedding('multilingual-cased') with adam for only 5 epochs. Even after 5 epochs, the accuracy exceeded 98% and one epoch takes only 40 minutes. So, I'm positive that the latter is better in terms of accuracy and computational power.  But I got one problem. When I predicted with the second one, I got the attribute error. My issue is  [here](https://github.com/flairNLP/flair/issues/1610).
